src/spec/services/s3client.spec.coffee

```diff
@@ -124,25 +124,43 @@ describe 'S3Client', ->
         dst: destination
         width: 200
       spyOn(easyimage, 'resize').andCallFake -> easyimagePromise
-      spyOn(Compress, 'compressImage')
-      spyOn(@s3client, 'putFile').andCallFake -> new Promise (resolve, reject) -> resolve()
 
     afterEach ->
       fs.unlinkAsync path + "/foo"
       fs.unlinkAsync destination
 
     it 'should call compress function', (done) ->
-
+      spyOn(Compress, 'compressImage')
+      spyOn(@s3client, 'putFile').andCallFake -> new Promise (resolve, reject) -> resolve()
       @s3client.resizeCompressAndUploadImages filesList, description, path, true
       .then ->
         expect(Compress.compressImage).toHaveBeenCalled()
       .then -> done()
       .catch (err) -> done(err)
 
     it 'shouldn\'t call compress function', (done) ->
-
+      spyOn(Compress, 'compressImage')
+      spyOn(@s3client, 'putFile').andCallFake -> new Promise (resolve, reject) -> resolve()
       @s3client.resizeCompressAndUploadImages filesList, description, path, false
       .then ->
         expect(Compress.compressImage).not.toHaveBeenCalled()
       .then -> done()
       .catch (err) -> done(err)
+
+    it 'should call put function without \"extends\" paramterer', (done) ->
```

Comment:
make the test description more clear.
"should call put without cache control headers"